test/rules/assertions/checkTagNames.js

```diff
@@ -674,6 +972,95 @@ export default {
     },
   ],
   valid: [
+    {
+      code: `
+        /** @abstract */
+        let a;
+      `,
+      filename: 'file.d.ts',
+      options: [
+        {
+          typed: true,
+        },
+      ],
+    },
+    {
+      code: `
+        /** @abstract */
+        declare let a;
+      `,
+      filename: 'file.d.ts',
+      options: [
+        {
+          typed: true,
+        },
+      ],
+      parser: require.resolve('@typescript-eslint/parser'),
+    },
+    {
+      code: `
+        /** @abstract */
+        { declare let a; }
+      `,
+      filename: 'file.d.ts',
+      options: [
+        {
+          typed: true,
+        },
+      ],
+      parser: require.resolve('@typescript-eslint/parser'),
+    },
+    {
+      code: `
+        function test() {
+          /** @abstract */
+          declare let a;
+        }
+      `,
+      filename: 'file.d.ts',
+      options: [
+        {
+          typed: true,
+        },
+      ],
+      parser: require.resolve('@typescript-eslint/parser'),
+    },
+    {
+      code: `
+        /** @abstract - default */
+        let a;
+      `,
+      options: [
+        {
+          typed: true,
+        },
+      ],
+    },
+    {
+      code: `
+        /** @template name */
+        let a;
+      `,
+      options: [
+        {
+          definedTags: [
+            'template',
+          ],
+          typed: true,
+        },
```

Comment:
This is fine, but I think I'd prefer here the following over `definedTags`:

```js
settings: {
        jsdoc: {
          mode: 'typescript',
        },
      },
```

This should better telegraph the desired approach when using TypeScript (enabling the mode or using the typescript-eslint parser which would also work here). One generally shouldn't need to configure `definedTags`.

Reply:
Err, this stuff predates me I think. Is there a change to the code added in this PR you're asking for?

Reply:
There may be another example like this that we do want to keep for coverage, but I'd rather not add it. Per the diffs, this looks like it is a new addition, esp. since it has `typed: true` on it. I guess you added it because without anything else, the rule will complain, but I'd just prefer we add the `settings` instead.

Reply:
I'm only referring to this example just above:

```js
{
      code: `
        /** @template name */
        let a;
      `,
      options: [
        {
          definedTags: [
            'template',
          ],
          typed: true,
        },
```

Reply:
Ah got it, thank you for the explanation. Sorry this took me a bit to parse 😄.

What finally clicked was removing `definedTags: ['template']` from the example. Then we get `Invalid JSDoc tag name "template".`. So, yes, these valid tags should be added to the list of valid ones if we're in `typed: true` mode! 💡

Reply:
Ok, I'd think `settings.jsdoc.mode` set to "typescript" would do the trick, as that should be set to "typescript" anyways for TS users (or auto-detected by use of `@typescript-eslint/parser` as parser, and although you could add the tag for `typed: true`, maybe you should keep `@template` disallowed there without the `mode`/parser in case someone is using some other type annotation system and wants to stick to pure JSDoc syntax?